### PR TITLE
ArchivesSpaceClient: escape spaces in searches

### PR DESCRIPTION
**Pull request by mistydemeo**
Fixes #9434.

### CODE REVIEW

**Comment by Hwesta** [comment]
Do we need to escape this for all queries, eg `count_collections`?  Are there other characters we have to escape?

**Comment by mistydemeo** [comment]
> Do we need to escape this for all queries, eg `count_collections`?

Yes; I'll add that.

> Are there other characters we have to escape?

Possibly, yeah. I need to look this up in the Solr guide.

**Comment by mistydemeo** [comment]
Looks like, after doing some research, there are a) many fields to escape, and b) the escaping rules are different for the `title` and `identifier` fields. The current revision is working as hoped, though.

File: tests/test_archivesspace_client.py [hunk 1]
```diff
@@ -393,3 +402,14 @@ def test_contentless_notes():
...
+
+
+def test_escaping_solr_queries():
+    client = ArchivesSpaceClient(**AUTH)
+    query = '"quotes"'
+    # Test escaping single characters
+    assert client._escape_solr_query(query, field='identifier') == '\\"quotes\\"'
+    assert client._escape_solr_query(query, field='title') == '\\\\"quotes\\\\"'
+    # And double characters, which require only one set of escape tokens
+    assert client._escape_solr_query('&&test', field='identifier') == '\\&&test'
```
Comment (Hwesta) [inline tests/test_archivesspace_client.py]: This should be the same as `r'\&&test'`, which might be easier to read (only one layer of escaping).

**Comment by Hwesta** [comment]
There are far too many layers of things being escaped here. Glad for the tests.

:+1:

File: tests/test_archivesspace_client.py (hunk 1 above)
Comment (mistydemeo) [inline tests/test_archivesspace_client.py]: Good idea; updated all of them to use `r''` strings.

**Pull request merged**